src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/AsyncAnalyzerUtilities.cs

```diff
@@ -0,0 +1,110 @@
+            TaskOfTTypeSymbol = compilation.GetTypeByMetadataName(typeof(Task<>).FullName);
+            ValueTaskTypeSymbol = compilation.GetTypeByMetadataName(typeof(ValueTask).FullName);
+            ValueTaskOfTTypeSymbol = compilation.GetTypeByMetadataName(typeof(ValueTask<>).FullName);
+            AsyncDisposableSymbol = compilation.GetTypeByMetadataName(typeof(IAsyncDisposable).FullName);
+            AsyncEnumerableOfTTypeSymbol = compilation.GetTypeByMetadataName(typeof(IAsyncEnumerable<>).FullName);
+            TaskAsyncEnumerableExtensionsSymbol = compilation.GetTypeByMetadataName(typeof(TaskAsyncEnumerableExtensions).FullName);
+        }
+
+        public bool IsConfigureAwait(SyntaxNode node) =>
+            node is InvocationExpressionSyntax invocation &&
+            invocation.Expression is MemberAccessExpressionSyntax memberAccess &&
+            memberAccess.Name.Identifier.Text == nameof(Task.ConfigureAwait);
+
+        public bool IsConfigureAwait(IMethodSymbol method)
+        {
+            if (method.Name == nameof(Task.ConfigureAwait) && method.Parameters.Length == 1 && IsTaskType(method.ReceiverType))
+            {
+                return true;
+            }
+
+            if (method.Name == nameof(TaskAsyncEnumerableExtensions.ConfigureAwait) && method.Parameters.Length == 2 && Equals(method.ReceiverType, TaskAsyncEnumerableExtensionsSymbol))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsAsyncDisposableType(ITypeSymbol type) 
+            => ImplementsInterface(type as INamedTypeSymbol, AsyncDisposableSymbol);
+
+        public bool IsAsyncEnumerableType(ITypeSymbol type) 
+            => ImplementsInterface(type as INamedTypeSymbol, AsyncEnumerableOfTTypeSymbol);
+
+        private static bool ImplementsInterface(INamedTypeSymbol type, INamedTypeSymbol candidate) 
+        {
+            if (type == null)
+            {
+                return false;
+            }
+
+            if (Equals(candidate, type.ConstructedFrom))
+            {
+                return true;
+            }
+
+            return type.AllInterfaces.Any(i => Equals(candidate, i.ConstructedFrom));
+        }
+
+        public bool IsTaskType(ITypeSymbol type)
+        {
+            if (type == null)
+            {
+                return false;
+            }
+
+            if (type.Equals(TaskTypeSymbol) || type.Equals(ValueTaskTypeSymbol))
+            {
+                return true;
+            }
+
+            if (type is INamedTypeSymbol namedType && namedType.IsGenericType) 
+            {
+                var genericType = namedType.ConstructedFrom;
+                if (Equals(genericType, TaskOfTTypeSymbol) || Equals(genericType, ValueTaskOfTTypeSymbol)) 
+                {
+                    return true;
+                }
+            }
+
+            if (type.TypeKind == TypeKind.Error)
```

Comment:
Interesting, when does this happen?

Reply:
For example, when type can't be resolved.